setup.py

```diff
@@ -1,84 +1,138 @@
+        Version="3.0.10" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventHubs"
+        Version="5.0.0" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid"
+        Version="3.1.0" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage"
+        Version="4.0.5" />
+      <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.ServiceBus"
+        Version="4.2.1" />
+      <PackageReference
+        Include="Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator"
+        Version="1.1.3" />
+   </ItemGroup>
 </Project>
 """
 
 
 NUGET_CONFIG = """\
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <configuration>
-  <packageSources>
-    <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
-    <add key="azure_app_service"
-         value="https://www.myget.org/F/azure-appservice/api/v2" />
-    <add key="azure_app_service_staging"
-         value="https://www.myget.org/F/azure-appservice-staging/api/v2" />
-    <add key="buildTools"
-         value="https://www.myget.org/F/30de4ee06dd54956a82013fa17a3accb/" />
-    <add key="AspNetVNext"
-         value="https://www.myget.org/F/aspnetcore-dev/api/v3/index.json" />
-  </packageSources>
+   <packageSources>
+      <add key="nuget.org"
+        value="https://www.nuget.org/api/v2/" />
+      <add key="azure_app_service"
+        value="https://www.myget.org/F/azure-appservice/api/v2" />
+      <add key="azure_app_service_staging"
+        value="https://www.myget.org/F/azure-appservice-staging/api/v2" />
+      <add key="buildTools"
+        value="https://www.myget.org/F/30de4ee06dd54956a82013fa17a3accb/" />
+      <add key="AspNetVNext"
+        value="https://www.myget.org/F/aspnetcore-dev/api/v3/index.json" />
+   </packageSources>
 </configuration>
 """
 
 
+CLASSIFIERS = [
+    "Development Status :: 5 - Production/Stable",
+    'Programming Language :: Python',
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX",
+    "Operating System :: MacOS :: MacOS X",
+    "Environment :: Web Environment",
+    "License :: OSI Approved :: MIT License",
+    "Intended Audience :: Developers",
+]
+
+
+PACKAGES = [
+    "azure_functions_worker",
+    "azure_functions_worker.protos",
+    "azure_functions_worker.protos.identity",
+    "azure_functions_worker.protos.shared",
+    "azure_functions_worker.bindings",
+    "azure_functions_worker.bindings.shared_memory_data_transfer",
+    "azure_functions_worker.utils",
+    "azure_functions_worker._thirdparty"
+]
+
+
+INSTALL_REQUIRES = [
+    "grpcio~=1.43.0",
+    "grpcio-tools~=1.43.0",
+    "protobuf~=3.19.3",
+    "azure-functions==1.8.0"
```

Comment:
Will movings az func from extra_requires to install_requires cause any dependency isolation issues?

Reply:
This will ensure that `azure-functions` get installed always.

Reply:
If you take a look at `nix_deps.sh`, referred in `nix_env_gen.yml` - it installs the `azure.functions` explicitly in `$BUILD_SOURCESDIRECTORY/deps` and that gets copied in `nix_env_gen.yml`. Thus, the `setup.py` is essentially useless when it comes to our packaging.

- `nix_env_gen.yml`
```yaml
- task: ShellScript@2
  inputs:
    disableAutoCwd: true
    scriptPath: 'pack/scripts/nix_deps.sh'
...
- task: CopyFiles@2
  inputs:
    sourceFolder: '$(Build.SourcesDirectory)/deps'
    contents: |
      **
      !grpc_tools/**/*
      !grpcio_tools*/*
    targetFolder: '$(Build.ArtifactStagingDirectory)'
```

- `nix_deps.sh`
```bash
#!/bin/bash

python -m venv .env
source .env/bin/activate
python -m pip install --upgrade pip

python -m pip install .

python -m pip install . azure-functions --no-compile --target "$BUILD_SOURCESDIRECTORY/deps"
```

Reply:
Looking at previous and new behavior - the only thing now changes is the time azure.functions is installed. We technically can remove `python -m pip install . azure-functions --no-compile --target "$BUILD_SOURCESDIRECTORY/deps"` being done explicitly. No need to do that in this PR but one possibility

![image](https://user-images.githubusercontent.com/1047040/152521654-e1ae35b0-45a7-4e98-89c0-8427a582bc6d.png)